File: Assignment 1/Problem8.py
def merge(intervals):
    intervals.sort(key=lambda x: x[0])
    merged = []

    for interval in intervals:
        if not merged or interval[0] > merged[-1][1]:
            merged.append(interval)
        else:
            merged[-1][1] = max(merged[-1][1], interval[1])

    return merged

intervals = [[1, 3], [2, 6], [8, 10], [15, 18]]
merged_intervals = merge(intervals)
print(merged_intervals)

File: ShaliniKumari_BanasthaliVidyapith/Arrays/Problem8.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
vector<vector<int>> merge(vector<vector<int>>& intervals) {
    sort(intervals.begin(),intervals.end());
    int n=intervals.size();
    vector<vector<int>>ans;
    for(int i=0;i<n;i++){
        if(ans.empty()){
            ans.push_back(intervals[i]);
        }else{
            vector<int>&a=ans.back();
            int y=a[1];
            if(intervals[i][0]<=y){
               a[1]=max(intervals[i][1],y);
            }else{
                ans.push_back(intervals[i]);
            }
        }
    }
    return ans;
}
int main() {
    vector<vector<int>> intervals= {{1,3},{2,6},{8,10},{15,18}};
    vector<vector<int>> answer = merge(intervals);
    for (const vector<int>& interval : answer) {
        cout << "[" << interval[0] << ", " << interval[1] << "] ";
    }
    cout << endl;
}